src/pages/gallery/[filename].astro

```diff
@@ -78,5 +78,10 @@ const isMobile = mobileUserAgent(Astro.request.headers.get('user-agent') ?? '')
 				}}
 			/>
 		)}
+
+		<script>
+			// There is no clear guidance there when it comes to handling device and browser-specific differentiations on mobile browsers,as such we need to do a custom implementation of vh
+			document.documentElement.style.setProperty('--vh', `${window.innerHeight * 0.01}px`);
```

Comment:
Okay so as I stated before (adapted from slack dms)

In mobile browsers since the page is dynamic, vh = window.outerHeight  = device height as to not flicker. Here I do the window.innerHeight / window.innerWidth so I can get the real VH/VW values and they get scaled between JS and CSS. The * 0.01 is done to get the 1/100 of the screen so I can scale respectively by 1-100 units